src/app/view/components/_common/dropdown-component/dot-dropdown.component.spec.ts

```diff
@@ -0,0 +1,150 @@
+    de: DebugElement,
+    comp: DotDropdownComponent
+) {
+    elem.nativeElement.click();
+    hostFixture.detectChanges();
+    const content = de.query(By.css('.dropdown-content'));
+    expect(content).toBeTruthy();
+    expect(elem).toBeTruthy();
+    expect(comp.toggle.emit).toHaveBeenCalledWith(true);
+}
+
+function executeDisabled(elem: DebugElement, de: DebugElement) {
+    elem.nativeElement.click();
+    tick(1);
+    const content = de.query(By.css('.dropdown-content'));
+    expect(content).toBeFalsy();
+    expect(elem).toBeTruthy();
+}
+
+describe('DotDropdownComponent', () => {
+    let hostFixture: ComponentFixture<DotTestHostComponent>;
+    let hostDe: DebugElement;
+    let hostComp: DotTestHostComponent;
+
+    let comp: DotDropdownComponent;
+    let de: DebugElement;
+
+    beforeEach(() => {
+        DOTTestBed.configureTestingModule({
+            declarations: [
+                DotDropdownComponent,
+                MockGravatarComponent,
+                MockDotIconButtonComponent,
+                DotTestHostComponent
+            ],
+            imports: [BrowserAnimationsModule]
+        });
+
+        hostFixture = DOTTestBed.createComponent(DotTestHostComponent);
+        hostComp = hostFixture.debugElement.componentInstance;
+        hostDe = hostFixture.debugElement;
+
+        de = hostDe.query(By.css('dot-dropdown-component'));
+        comp = de.componentInstance;
+        hostComp.icon = 'icon';
+        hostComp.title = 'test';
+        hostComp.gravatar = 'test@test.com';
+    });
+
+    describe('Enabled', () => {
+        let button: DebugElement;
+        let titleButton: DebugElement;
+        let gravatar: DebugElement;
+
+        beforeEach(() => {
+            spyOn(comp.toggle, 'emit');
+            hostComp.disabled = false;
+            hostFixture.detectChanges();
+            button = de.query(By.css('dot-icon-button'));
+            titleButton = de.query(By.css('button'));
+            gravatar = de.query(By.css('dot-gravatar'));
+        });
+
+        it(`should dot-icon button be displayed & emit`, () => {
+            executeEnabled(button, hostFixture, de, comp);
+            expect(button.attributes.disabled).toBe(null);
+        });
+
+        it(`should title button be displayed & emit`, () => {
+            executeEnabled(titleButton, hostFixture, de, comp);
+        });
+
+        it(`should dot-gravatar be displayed & emit`, () => {
+            executeEnabled(gravatar, hostFixture, de, comp);
+        });
+    });
+
+    describe('Disabled', () => {
+        let button: DebugElement;
+        let titleButton: DebugElement;
+
+        beforeEach(() => {
+            spyOn(comp.toggle, 'emit');
+            hostComp.disabled = true;
+            hostFixture.detectChanges();
+            button = de.query(By.css('dot-icon-button'));
+            titleButton = de.query(By.css('button'));
+        });
+
+        it(`should dot-icon button not be displayed --> null`, fakeAsync(() => {
```

Comment:
Why do you need `fakeAsync` here?

Reply:
to reflect wait that the changes get reflected on the component `.dropdown-content`

Reply:
I just run it with no `fakeAsync` and worked.